s2i-setup

```diff
@@ -5,9 +5,9 @@ set -eo pipefail
 # We need to setup these directories before the actual assembly process
 # because odo copies into some of these directories before running assemble
 
+# source directory from other s2i images
+mkdir -p ${ODO_S2I_SRC_BIN_PATH}
```

Comment:
This needs to be `mkdir -p ${ODO_S2I_SRC_BIN_PATH}/src` otherwise odo will fail to push code

Reply:
Hm... Actually my statement above is not correct. It's needed for java but no nodejs. So I need to look into this more

Reply:
The situation here is that the way `ODO_S2I_SRC_BIN_PATH` is currently populated, it's not guaranteed to contain `/src` suffix.

So I propose that in odo proper we add this change (in `occlient.GetS2IPathsFromBuilderImg`):

```
	if s2iSrcOrBinPath == "" {
		// In cases like nodejs builder image, where there is no concept of binary and sources are directly run, use destination as source
		s2iSrcOrBinPath = getS2ILabelValue(dimdr.ContainerConfig.Labels, S2IDeploymentsDir)
	}
```

That way my comments above are not needed

Reply:
Forget my comment above. The correct way to do things that works for all s2i images (at least the ones that I tested) is to use:

```
# ensure that the source directory is exists
mkdir -p ${ODO_S2I_SRC_BIN_PATH}
touch ${ODO_S2I_SRC_BIN_PATH}/.dummy

# some s2i images additionally expect /tmp/src to exist
if [ "${ODO_S2I_SRC_BIN_PATH}" != "/tmp/src" ]; then
  mkdir -p /tmp/src
  touch /tmp/src/.dummy
fi
```